tests/lib/jsontests.icn

```diff
@@ -243,3 +246,48 @@ procedure objects_equal(x,y)
       }  
    return
 end
+
+
+procedure schema_test()
+
+good :=
+"{ _
+  \"firstName\": \"John\", _
+  \"lastName\": \"Doe\", _
+  \"age\": 22 _
+}"
+
+bad :=
+"{ _
+  \"firstName\": \"John\", _
+  \"lastName\": \"Doe\", _
+  \"age\": -2 _
+}"
+
+
+schema :=
+"{ _
+  \"title\": \"Person\", _
+  \"type\": \"object\", _
+  \"properties\": { _
+    \"firstName\": { _
+      \"type\": \"string\", _
+    }, _
+    \"lastName\": { _
+      \"type\": \"string\", _
+    }, _
+    \"age\": { _
+      \"type\": \"integer\", _
+      \"minimum\": 0 _
+    } _
+  } _
+}"
+
+
+   v := jss_validate(good, schema)
+   write("schema test valid:", if v=="" then  "SUCCESS" else "FAILURE " )
+
+   v := jss_validate(bad, schema)
+   write("schema test invalid:", if v~=="" then  "SUCCESS" else "FAILURE " )
```

Comment:
The helper function called by jss_validate has a large number of cases which are not explored by these two tests. Would it be possible to increase the coverage?  I'm not asking for 100% (or anywhere near) but this exercises only a tiny fraction of the code.

Reply:
of course!